js/packages/teams-ai/src/InputFileDownloader.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @module teams-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TurnContext } from 'botbuilder';
+import { TurnState } from './TurnState';
+
+/**
+ * A plugin responsible for downloading files relative to the current users input.
+ * @template TState Optional. Type of application state.
+ */
+export interface InputFileDownloader<TState extends TurnState = TurnState> {
+    /**
+     * Download any files relative to the current users input.
+     * @param context Context for the current turn of conversation.
+     * @param state Application state for the current turn of conversation.
+     */
+    downloadFiles(context: TurnContext, state: TState): Promise<InputFile[]>;
+}
+
+/**
+ * A file sent by the user to the bot.
+ */
+export interface InputFile {
+    /**
+     * The downloaded content of the file.
+     */
+    content: Buffer;
+
+    /**
+     * The content type of the file.
+     */
+    contentType: string;
```

Comment:
Are we limiting the types at all? what format does this need to be in? Mimetype?

Reply:
I wanted to make this general purpose so that it could be used beyond just supporting vision. I wasn't going to restrict the types of files that can be downloaded.